src/Stratis.Bitcoin.Tests/Connection/ConnectionManagerTest.cs

```diff
@@ -0,0 +1,107 @@
+            endpointB = new NodeServerEndpoint(new IPEndPoint(IPAddress.Parse("0.0.0.0"), 16178), false);
+            connectionManagerSettings.Listen = new List<NodeServerEndpoint>() { endpointB};
+            connectionManagerSettings.Port = 16178;
+            Assert.True(endpointA.CanBeMappedTo(connectionManagerSettings.Listen, out endpointOut));
+
+            // 127.0.0.1:16178 != 0.0.0.0:44556
+            endpointA = new NodeServerEndpoint(new IPEndPoint(IPAddress.Parse("127.0.0.1"), 16178), false);
+            endpointB = new NodeServerEndpoint(new IPEndPoint(IPAddress.Parse("0.0.0.0"), 44556), false);
+            connectionManagerSettings.Listen = new List<NodeServerEndpoint>() { endpointB };
+            connectionManagerSettings.Port = 44556;
+            Assert.False(endpointA.CanBeMappedTo(connectionManagerSettings.Listen, out endpointOut));
+
+            // 0.0.0.0:16178 == 127.0.0.1:16178
+            endpointA = new NodeServerEndpoint(new IPEndPoint(IPAddress.Parse("0.0.0.0"), 16178), false);
+            endpointB = new NodeServerEndpoint(new IPEndPoint(IPAddress.Parse("127.0.0.1"), 16178), false);
+            connectionManagerSettings.Listen = new List<NodeServerEndpoint>() { endpointB };
+            connectionManagerSettings.Port = 16178;
+            Assert.True(endpointA.CanBeMappedTo(connectionManagerSettings.Listen, out endpointOut));
+
+            // IPV4: 0.0.0.0:16178 == 127.0.0.2:16178
+            endpointA = new NodeServerEndpoint(new IPEndPoint(IPAddress.Parse("0.0.0.0"), 16178), false);
+            endpointB = new NodeServerEndpoint(new IPEndPoint(IPAddress.Parse("127.0.0.2"), 16178), false);
+            connectionManagerSettings.Port = 16178;
+            connectionManagerSettings.Listen = new List<NodeServerEndpoint>() { endpointB };
+            Assert.True(endpointA.CanBeMappedTo(connectionManagerSettings.Listen, out endpointOut));
+
+            // IPV6: [::]:16178 == [::1]:16178
+            endpointA = new NodeServerEndpoint(new IPEndPoint(IPAddress.Parse("[::]"), 16178), false);
+            endpointB = new NodeServerEndpoint(new IPEndPoint(IPAddress.Parse("[::1]"), 16178), false);
+            connectionManagerSettings.Port = 16178;
+            connectionManagerSettings.Listen = new List<NodeServerEndpoint>() { endpointB };
+            Assert.True(endpointA.CanBeMappedTo(connectionManagerSettings.Listen, out endpointOut));
+
+            // IPV4: 0.0.0.0:16178 == 127.0.0.2:999
+            endpointA = new NodeServerEndpoint(new IPEndPoint(IPAddress.Parse("0.0.0.0"), 16178), false);
+            endpointB = new NodeServerEndpoint(new IPEndPoint(IPAddress.Parse("127.0.0.2"), 999), false);
+            connectionManagerSettings.Listen = new List<NodeServerEndpoint>() { endpointB };
+            connectionManagerSettings.Port = 999;
+            Assert.False(endpointA.CanBeMappedTo(connectionManagerSettings.Listen, out endpointOut));
+
+            // IPV6: [::]:16178 != [::2]:999
```

Comment:
Could add equivalent test for 127.0.0.1:16178 != 0.0.0.0:16178 in IPv6 format
// IPV6: [::]:16178 != [::1]:16178